models/Import/Service/QtiResultXmlImporter.php

```diff
@@ -0,0 +1,191 @@
+use qtism\data\storage\xml\XmlStorageException;
+use taoQtiTest_models_classes_QtiTestService;
+use taoResultServer_models_classes_WritableResultStorage;
+
+class QtiResultXmlImporter
+{
+    private Ontology $ontology;
+    private ResultServerService $resultServerService;
+    private QtiResultXmlFactory $qtiResultXmlFactory;
+    private QtiResultParser $qtiResultParser;
+    private taoQtiTest_models_classes_QtiTestService $qtiTestService;
+    private DeliveryExecutionService $deliveryExecutionService;
+
+    public function __construct(
+        Ontology $ontology,
+        ResultServerService $resultServerService,
+        QtiResultXmlFactory $qtiResultXmlFactory,
+        QtiResultParser $qtiResultParser,
+        taoQtiTest_models_classes_QtiTestService $qtiTestService,
+        DeliveryExecutionService $deliveryExecutionService
+    ) {
+        $this->ontology = $ontology;
+        $this->resultServerService = $resultServerService;
+        $this->qtiResultXmlFactory = $qtiResultXmlFactory;
+        $this->qtiResultParser = $qtiResultParser;
+        $this->qtiTestService = $qtiTestService;
+        $this->deliveryExecutionService = $deliveryExecutionService;
+    }
+
+    /**
+     * @throws ImportResultException
+     * @throws XmlStorageException
+     * @throws common_exception_Error
+     * @throws common_exception_InvalidArgumentType
+     * @throws common_exception_NotFound
+     * @throws common_exception_NotImplemented
+     * @throws core_kernel_persistence_Exception
+     */
+    public function importByResultInput(ImportResultInput $input): void
+    {
+        $this->importQtiResultXml(
+            $input->getDeliveryExecutionId(),
+            $this->qtiResultXmlFactory->createByImportResult($input)
+        );
+    }
+
+    /**
+     * @throws ImportResultException
+     * @throws XmlStorageException
+     * @throws common_exception_NotFound
+     * @throws common_exception_Error
+     * @throws common_exception_InvalidArgumentType
+     * @throws common_exception_NotImplemented
+     * @throws core_kernel_persistence_Exception
+     */
+    public function importQtiResultXml(
+        string $deliveryExecutionId,
+        string $xmlContent
+    ): void {
+        $resultMapper = $this->qtiResultParser->parse($xmlContent);
+
+        $deliveryExecution = $this->deliveryExecutionService->getDeliveryExecution($deliveryExecutionId);
+        $delivery = $deliveryExecution->getDelivery();
+
+        $resultStorage = $this->resultServerService->getResultStorage();
+
+        if (!$resultStorage instanceof taoResultServer_models_classes_WritableResultStorage) {
+            throw new ImportResultException(
+                sprintf(
+                    'ResultStorage must be an instance of %s. Instance of %s provided',
+                    taoResultServer_models_classes_WritableResultStorage::class,
+                    get_class($resultStorage)
+                )
+            );
+        }
```

Comment:
You can call it at the beginning of the method to avoid unnecessary calls

Reply:
Done, will be pushed soon!